### PR TITLE
[WIP][Relay] XXXNode::make to constructor

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
@tqchen, @jroesch 
Sample change for #5069

### CODE REVIEW

**Comment by ANSHUMAN87** [comment]
I have only made changes for IfNode, to get your valuable opinion. Once it is through, i can follow same template for others.

I had a dilemma of the format of constructor: 
    1:> If(IfNode(Args))
    2:> If(Args) --> Ifnode(Args)

Currently implemented as per 1. 
Eagerly waiting for your valuable feedback. TIA!

File: include/tvm/relay/expr.h [hunk 1]
```diff
@@ -284,18 +285,28 @@ class Let : public Expr {
 class If;
 /*! \brief container of If */
 class IfNode : public ExprNode {
- public:
   /*! \brief The condition */
-  Expr cond;
+  Expr _cond;
```
Comment (zhiics) [inline include/tvm/relay/expr.h]: Don't need this change to keep consistent to the used naming style.

**Comment by tqchen** [comment]
Thanks for taking the initiative. Given that the change could touch quite a few places. please also reply to the corresponding issue. 

 The particular change we intended is not to introduce member accessors and constructor in the Node class, but instead move the constructor from XXNode::make directly to the constructor of the reference class

Perhaps @zhiics can also followup.

**Comment by zhiics** [comment]
Yeah, I can send a PR later if @ANSHUMAN87 doesn't mind.

**Comment by ANSHUMAN87** [comment]
Thanks @tqchen , @zhiics . Closing as zhiics has handled as part of PR(#5128)

**Pull request closed without merging**